packages/web3-eth/src/web3_subscriptions.ts

```diff
@@ -78,8 +88,8 @@ export class NewHeadsSubscription extends Web3Subscription<
 		return ['newHeads'] as ['newHeads'];
 	}
 
-	protected _processSubscriptionResult(data: BlockOutput) {
-		this.emit('data', data);
+	protected _processSubscriptionResult(data: BlockInput) {
+		this.emit('data', outputBlockFormatter(data));
```

Comment:
The schema for block header will be different from https://github.com/ChainSafe/web3.js/blob/acf0950ebcdc6ea0159175e9e8760420bd6aa8aa/packages/web3-eth/src/schemas.ts#L227

as blockheader will have no `size` field, and it will have `sha3Uncles`, and no transactions array.

Sample block header sub result:
```
{
    "jsonrpc": "2.0",
    "method": "eth_subscription",
    "params": {
        "subscription": "0x41000f5f2001aef6650a30e22c3d0d5dffb0f6a367cc",
        "result": {
            "number": "0xe72237",
            "hash": "0x5164d356f1dc9ef85679efaf5db2b75c5d0f5bd79f8273b06f1b9f9a50cd8649",
            "parentHash": "0x5ee2cfa04b08691aae25b42f7e7cdda0e827ecbc683787d6c06656a00d092c48",
            "sha3Uncles": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
            "logsBloom": "0xda6f6bc2c71015efe1f2afdcf649feb39879997d2df5a7d9f6db229b76abfe47957cfc67ccd9f0fe98b2dbeec32fdbedf665be82fb8fb4fb2d7fc95b7b3afffcfcfdffa0bf49987fabb1a27b2b6ffde6fbb75ec7afffd84cbcf2fd7ed716fbfdfbcc781a4fba320fcd6f36551feee88e1bf3aefeffb0e7d1a7cd9ef9ff6a516a562fbfec61ee346b9f6fba71bbd1e83607909df7a564f6ad2f70b1c78af1b8ebfbdb6d5b9be8ec6bfb976ad3b7f5bfedb8be7ed64ec98b6a3dab6fb39e4f6a7c8fd9e433dac76cfe7757c53fdabfddd5bb6cfeefc5fff8dcf28d7d9fb8f5736745dafecba92efddef5fedeea15f86e46a4f969e2e5efb6f3f359dadb93cb79e7",
            "transactionsRoot": "0x75c20c8f88aed30e99ae4f8090f910a8de2521b1be24695de7028abbe45e3783",
            "stateRoot": "0x0d8f4f16dd6215b82e8340b5563b2ecbf84bb426f2f0b26f5515844de6963061",
            "receiptsRoot": "0x786e919090af307568e0da2ab6d34abbe209b73a35b1476a982fba5fb9ce2e55",
            "miner": "0xea674fdde714fd979de3edf0f56aa9716b898ec8",
            "difficulty": "0x28ee03c4fe8e82",
            "extraData": "0x617369612d65617374322d3133",
            "gasLimit": "0x1ca35ef",
            "gasUsed": "0x1a80e77",
            "timestamp": "0x62d16df5",
            "baseFeePerGas": "0x8e1b181e8",
            "nonce": "0x95664170456cd550",
            "mixHash": "0xf9a1ff2e8b4963b976ce663680a1b5f6258bf68d49629db112fb319ab6205d5d"
        }
    }
}
```